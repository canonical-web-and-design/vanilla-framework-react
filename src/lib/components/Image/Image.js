import React from 'react';
import PropTypes from 'prop-types';
import './Image.css';

const Image = (props) => {
  let classArray = [];

  if (props.bordered) {
    classArray = [...classArray, 'p-image--bordered'];
  }
  if (props.shadowed) {
    classArray = [...classArray, 'p-image--shadowed'];
  }

  const classString = classArray.join(' ');

  return <img className={classString} src={props.src} alt={props.alt} />;
};

Image.defaultProps = {
  alt: '',
  bordered: false,
  shadowed: false,
};

Image.propTypes = {
  src: PropTypes.string.isRequired,
  alt: PropTypes.string,
  bordered: PropTypes.bool,
  shadowed: PropTypes.bool,
};

Image.displayName = 'Image';

export default Image;
