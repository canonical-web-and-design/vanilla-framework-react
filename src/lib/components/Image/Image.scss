@import 'vanilla-framework/scss/base';
@import 'vanilla-framework/scss/patterns_image';
@include vf-b-media;
@include vf-p-image;
