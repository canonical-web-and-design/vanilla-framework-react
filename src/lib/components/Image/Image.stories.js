import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';

import Image from './Image';

storiesOf('Image', module)
  .add('Default',
    withInfo('Default Image component.')(() => (
      <Image src="http://placekitten.com/g/300/300" alt="" />),
    ),
  )
  .add('Bordered',
    withInfo('Bordered Image component.')(() => (
      <Image bordered src="http://placekitten.com/g/300/300" alt="" />),
    ),
  )
  .add('Shadowed',
    withInfo('Image component with drop-shadow.')(() => (
      <Image shadowed src="http://placekitten.com/g/300/300" alt="" />),
    ),
  )
  .add('Bordered and shadowed',
    withInfo('Image component with both border and drop-shadow.')(() => (
      <Image bordered shadowed src="http://placekitten.com/g/300/300" alt="" />),
    ),
  );
