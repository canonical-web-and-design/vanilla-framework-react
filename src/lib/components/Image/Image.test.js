import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import Image from './Image';

describe('Image component', () => {
  it('should render a default Image correctly', () => {
    const image = ReactTestRenderer.create(
      <Image src="http://placekitten.com/g/300/300" alt="" />);
    const json = image.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render bordered prop correctly', () => {
    const image = ReactTestRenderer.create(
      <Image bordered src="http://placekitten.com/g/300/300" alt="" />);
    const json = image.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render shadowed prop correctly', () => {
    const image = ReactTestRenderer.create(
      <Image shadowed src="http://placekitten.com/g/300/300" alt="" />);
    const json = image.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render both bordered and shadowed props correctly', () => {
    const image = ReactTestRenderer.create(
      <Image bordered shadowed src="http://placekitten.com/g/300/300" alt="" />);
    const json = image.toJSON();
    expect(json).toMatchSnapshot();
  });
});
