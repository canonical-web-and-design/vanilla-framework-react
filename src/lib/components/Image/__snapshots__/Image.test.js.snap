// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image component should render a default Image correctly 1`] = `
<img
  alt=""
  className=""
  src="http://placekitten.com/g/300/300"
/>
`;

exports[`Image component should render bordered prop correctly 1`] = `
<img
  alt=""
  className="p-image--bordered"
  src="http://placekitten.com/g/300/300"
/>
`;

exports[`Image component should render both bordered and shadowed props correctly 1`] = `
<img
  alt=""
  className="p-image--bordered p-image--shadowed"
  src="http://placekitten.com/g/300/300"
/>
`;

exports[`Image component should render shadowed prop correctly 1`] = `
<img
  alt=""
  className="p-image--shadowed"
  src="http://placekitten.com/g/300/300"
/>
`;
